File: problemSet/599A_Patrick_and_Shopping.py
"""
Codeforces Round #332

Problem 599 A. Patrick and Shopping

@author yamaton
@date 2015-11-20
"""

import sys


def solve(d1, d2, d3):
    return min(d1 + d2 + d3, 2*(d1 + d2), 2*(d1 + d3), 2*(d2 + d3))


def p(*args, **kwargs):
    return print(*args, file=sys.stderr, **kwargs)


def main():
    [d1, d2, d3] = map(int, input().strip().split())
    result = solve(d1, d2, d3)
    print(result)


if __name__ == '__main__':
    main()

File: problemSet/599A_Patrick_and_Shopping.cpp
/*
 * Codeforces Round #332 (Div. 2)
 *
 * Problem 599 A. Patrick and Shopping
 *
 * author: yamaton
 * date: 2015-11-20
 *
 */

#include <iostream>
#include <vector>

using namespace std;

int solve(int d1, int d2, int d3) {
    vector<int> v{d1 + d2 + d3,
                  2 * (d1 + d2),
                  2 * (d2 + d3),
                  2 * (d1 + d3)};
    return *min_element(v.begin(), v.end());
}


int main() {
    int d1, d2, d3;
    cin >> d1;
    cin >> d2;
    cin >> d3;
    int ans = solve(d1, d2, d3);
    cout << ans << endl;
}